kubernetes-kit-starter/src/main/java/com/vaadin/kubernetes/starter/sessiontracker/SessionSerializer.java

```diff
@@ -313,6 +313,8 @@ private void queueSerialization(String sessionId, Duration timeToLive,
                 }).whenComplete((unused, error) -> {
                     pending.remove(sessionId);
                     if (error != null) {
+                        backendConnector.markSerializationFailed(clusterKey,
```

Comment:
nit: `error` will most likely be a `CompletionException`. In this case, it would be nice to unwrap and forward the real cause to the connector.

Reply:
Added the suggested check to unwrap the `CompletionException`.